client/src/language/middlewareHover.ts

```diff
@@ -11,9 +11,13 @@ import { getEmbeddedLanguageDocPosition } from './utils'
 import { getFileContent } from '../lib/src/utils/files'
 
 export const middlewareProvideHover: HoverMiddleware['provideHover'] = async (document, position, token, next) => {
+  const nextResult = await next(document, position, token)
+  if (nextResult !== undefined) {
```

Comment:
so it will always use the hover results from our language server should they exist. I don't think it has any major side effects here. Still, it'd be nice to have a test for it.

Reply:
You're right, this needs tests. However, it will need integration tests, and the [branch adding integration tests](https://github.com/savoirfairelinux/vscode-bitbake/pull/79) is still blocked :/

I thought I could do without integration tests with [some changes](https://github.com/savoirfairelinux/vscode-bitbake/tree/refactor-move-embedded-documents-manager-to-client), but finally it does not help at all.

Reply:
The other PR can be merged if it doesn't introduce more errors on the debug launch configuration for integration tests.